ui/core/components/suggest_reforges_action.tsx

```diff
@@ -207,31 +207,45 @@ export class ReforgeOptimizer {
 	protected readonly sim: Sim;
 	protected readonly defaults: IndividualSimUI<any>['individualConfig']['defaults'];
 	protected getEPDefaults: ReforgeOptimizerOptions['getEPDefaults'];
-	protected _statCaps: Stats;
+	protected _statCaps: Stats = new Stats();
+	protected breakpointLimits: Stats = new Stats();
 	protected updateGearStatsModifier: ReforgeOptimizerOptions['updateGearStatsModifier'];
 	protected _softCapsConfig: StatCap[];
+	private useCustomEPValues = false;
+	private useSoftCapBreakpoints = true;
+	protected softCapBreakpoints: StatCap[] = [];
 	protected updateSoftCaps: ReforgeOptimizerOptions['updateSoftCaps'];
 	protected enableBreakpointLimits: ReforgeOptimizerOptions['enableBreakpointLimits'];
 	protected statTooltips: StatTooltipContent = {};
 	protected additionalSoftCapTooltipInformation: StatTooltipContent = {};
 	protected statSelectionPresets: ReforgeOptimizerOptions['statSelectionPresets'];
-	readonly includeGemsChangeEmitter = new TypedEvent<void>();
 	protected includeGems = false;
-	readonly includeEOTBPGemSocketChangeEmitter = new TypedEvent<void>();
 	protected includeEOTBPGemSocket = false;
-	readonly freezeItemSlotsChangeEmitter = new TypedEvent<void>();
 	protected freezeItemSlots = false;
-	protected frozenItemSlots = new Map<ItemSlot, boolean>();
```

Comment:
Changed this as we dont need a slot x boolean reference, a Set only containing the ItemSlot suffices.